server/src/test/java/org/opensearch/index/shard/SegmentReplicationIndexShardTests.java

```diff
@@ -207,6 +208,52 @@ public void testMergedSegmentReplicationWithZeroReplica() throws Exception {
         }
     }
 
+    @LockFeatureFlag(MERGED_SEGMENT_WARMER_EXPERIMENTAL_FLAG)
+    public void testCleanupRedundantPendingMergeSegment() throws Exception {
```

Comment:
this is really a good test for simulating the scenario you mentioned in PR description.